### PR TITLE
Naive handling for online video links

### PR DESCRIPTION
**Pull request by IguanasInPyjamas**
Works on links to the direct file. Doesn't handle youtube. Should we consider adding youtube-dl for those types of problems @MattAG

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -21,11 +21,15 @@ fn main() {
...
     //TODO Use the GNU online videos for unit testing purposes.
     // TODO Implement unit tests.
-
+    let mut filepath :std::string::String = "".to_string();
     //For use in unit tests.
-    //let test_path = ""https://www.freedesktop.org/software/gstreamer-sdk/data/media/sintel_cropped_multilingual.webm";
-
-    let filepath = uri::get_uri_from_path(path);
+    //let path = "https://www.freedesktop.org/software/gstreamer-sdk/data/media/sintel_cropped_multilingual.webm";
+    if path.contains("https"){
```
Comment (IguanasInPyjamas) [inline src/main.rs:27]: the naivety comes here, not all websites are https.

**Comment by MattAG** [comment]
> Works on links to the direct file. Doesn't handle youtube. Should we consider adding youtube-dl for those types of problems @MattAG 

Which URLs have you tried with Youtube?

File: src/main.rs (hunk 1 above)
Comment (MattAG) [reply src/main.rs:27]: Should, at the very least, also allow http.

Comment (IguanasInPyjamas) [reply src/main.rs:27]: On it

**Comment by IguanasInPyjamas** [comment]
> > Works on links to the direct file. Doesn't handle youtube. Should we consider adding youtube-dl for those types of problems @MattAG
> 
> Which URLs have you tried with Youtube? i.e. Have you tried embed links too?

Well, it's more that I know it probably won't work, As a YouTube link isn't linking only to the video. I'll test it though

**Comment by MattAG** [comment]
A quick search makes it look possible but it might require a GStreamer plugin.

File: src/main.rs (hunk 1 above)
Comment (IguanasInPyjamas) [reply src/main.rs]: Done

**Comment by IguanasInPyjamas** [comment]
> A quick search makes it look possible but it might require a GStreamer plugin.

It's good to think about for the future.

Also, @MattAG , I've restructured the file system as well as adding queuing, it works well but shuffle / loop needs to be implemented. Let me know your thoughts

**Comment by MattAG** [comment]
Added more comments now. I'm basically running off the rule that functions should be ~25 lines max. Obviously, there can be exceptions to this, where it makes sense, but it's a decent rule of thumb. If a function only does one thing, it can be unit tested.

**Review by MattAG — CHANGES_REQUESTED** [review]

File: src/main.rs [hunk 2]
```diff
@@ -9,137 +9,21 @@ pub mod stream_info;
...
-                if keys.contains(&Keycode::S){
-                    inputs::switch_subs(&mut stream_info);
-                }
-            }
+    if args.len() == 2{
+       player::player_loop(args);
+    } else {
+        for i in 1..(args.len()){
+            let new_arg:Vec<String> = vec![args[0].clone(),args[i].clone()];
+            player::player_loop(new_arg);
```
[the remark this answers is not in the document]
Comment (IguanasInPyjamas) [reply src/main.rs:25]: It destroys and then creates a new player yes, I need to experiment some more with uri properties to see if a more elegant solution can be found

File: src/main.rs [hunk 3]
```diff
@@ -9,137 +9,21 @@ pub mod stream_info;
...
-
-                if keys.contains(&Keycode::S){
-                    inputs::switch_subs(&mut stream_info);
-                }
-            }
+    if args.len() == 2{
+       player::player_loop(args);
+    } else {
+        for i in 1..(args.len()){
+            let new_arg:Vec<String> = vec![args[0].clone(),args[i].clone()];
```
[the remark this answers is not in the document]
Comment (IguanasInPyjamas) [reply src/main.rs:24]: 100%

**Review by MattAG — APPROVED** [review]
Gj

**Pull request merged**